server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeMetaWriter.java

```diff
@@ -0,0 +1,226 @@
+        long dataFilePointer,
+        long dataFileLength
+    ) throws IOException {
+
+        if (logger.isDebugEnabled()) {
+            long totalSizeInBytes = 0;
+
+            // header size
+            totalSizeInBytes += computeHeaderByteSize(compositeFieldType, starTreeField.getName());
+            // number of dimensions
+            totalSizeInBytes += Integer.BYTES;
+            // dimension field numbers
+            totalSizeInBytes += (long) starTreeField.getDimensionsOrder().size() * Integer.BYTES;
+            // metric count
+            totalSizeInBytes += Integer.BYTES;
+            // metric - metric stat pair
+            totalSizeInBytes += computeMetricEntriesSizeInBytes(metricAggregatorInfos);
+            // segment aggregated document count
+            totalSizeInBytes += Integer.BYTES;
+            // max leaf docs
+            totalSizeInBytes += Integer.BYTES;
+            // skip star node creation dimensions count
+            totalSizeInBytes += Integer.BYTES;
+            // skip star node creation dimensions field numbers
+            totalSizeInBytes += (long) starTreeField.getStarTreeConfig().getSkipStarNodeCreationInDims().size() * Integer.BYTES;
+            // data start file pointer
+            totalSizeInBytes += Long.BYTES;
+            // data length
+            totalSizeInBytes += Long.BYTES;
+
+            logger.debug("Star tree size in bytes : {}", totalSizeInBytes);
+        }
+
+        writeMetaHeader(metaOut, compositeFieldType, starTreeField.getName());
+        writeMeta(metaOut, writeState, metricAggregatorInfos, starTreeField, segmentAggregatedCount, dataFilePointer, dataFileLength);
+    }
+
+    /**
+     * Computes the byte size required to store the star-tree metric entry.
+     *
+     * @param metricAggregatorInfos the list of metric aggregator information
+     * @return the byte size required to store the metric-metric stat pairs
+     */
+    private static long computeMetricEntriesSizeInBytes(List<MetricAggregatorInfo> metricAggregatorInfos) {
+
+        long totalMetricEntriesSize = 0;
+
+        for (MetricAggregatorInfo metricAggregatorInfo : metricAggregatorInfos) {
+            totalMetricEntriesSize += metricAggregatorInfo.getField().getBytes(UTF_8).length;
+            totalMetricEntriesSize += metricAggregatorInfo.getMetricStat().getTypeName().getBytes(UTF_8).length;
+        }
+
+        return totalMetricEntriesSize;
+    }
+
+    /**
+     * Computes the byte size of the star-tree metadata header.
+     *
+     * @param compositeFieldType the composite field type of the star-tree field
+     * @param starTreeFieldName  the name of the star-tree field
+     * @return the byte size of the star-tree metadata header
+     */
+    private static int computeHeaderByteSize(CompositeMappedFieldType.CompositeFieldType compositeFieldType, String starTreeFieldName) {
+        // Magic marker (8), version (4), size of header (4)
+        int headerSizeInBytes = 16;
```

Comment:
Similar comment : Can we compute and keep track as and when we write the fields ?

Reply:
removed this compute. Using file pointers to track meta size